### PR TITLE
Experience Detail | Image carousel

### PR DESCRIPTION
**Pull request by Alvtag**
**Description**
Replaces imageView with a carouselView with indicator and snapping

**Issues that should be CLOSED by merge of this PR:**
TPP-49
https://guestlogix.atlassian.net/jira/software/projects/TPP/boards/4?label=Android&selectedIssue=TPP-49

### CODE REVIEW

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/adapters/ImageCarouselRecyclerViewAdapter.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Override
+    public int getItemCount() {
+        return (null != data) ? data.size() : 0;
+    }
+
+    public void setData(List<URL> data) {
+        this.data = data;
+        notifyDataSetChanged();
```
Comment (ata-n) [inline traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/adapters/ImageCarouselRecyclerViewAdapter.java]: should not call notify...its the callers responsibility to call notify when they are ready

Comment (ata-n) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/adapters/ImageCarouselRecyclerViewAdapter.java]: in fact I think its best to make this property immutable, well make it mutable if we need it

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    }
+
+    public CarouselView(Context context) {
+        super(context);
+        initView();
+    }
+
+
+    public void setImageURLList(List<URL> imageUrls) {
+        adapter.setData(imageUrls);
```
Comment (ata-n) [inline traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java]: this type of proxying should be avoided.

you should only set the adapter directly?

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    private void initView() {
+        recyclerView = (RecyclerView) inflate(getContext(), R.layout.view_carousel, null);
+        LayoutParams params = new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
+        recyclerView.setLayoutParams(params);
+        addView(recyclerView);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getContext(), RecyclerView.HORIZONTAL, false));
+        recyclerView.setAdapter(adapter);
+
+        // android.support.v7.widget.LinearSnapHelper is available without androidX, see
+        // http://androidappcoders.blogspot.com/2018/06/using-snaphelper-in-recyclerview.html
```
Comment (ata-n) [inline traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java]: +1

File: traveler_ui_kit/src/main/res/values/colors.xml [hunk 4]
```diff
@@ -16,4 +16,7 @@
     <color name="off_white">#EEEEEE</color>
     <color name="catalogOverlay">#40000000</color>
     <color name="dialogTitle">#262827</color>
-</resources>
\ No newline at end of file
+
+    <color name="catalog_item_detail_indicator_inactive">#505050</color>
+    <color name="catalog_item_detail_indicator_active">#EEEEEE</color>
```
[the remark this answers is not in the document]
Comment (Alvtag) [reply traveler_ui_kit/src/main/res/values/colors.xml]: will change to 
<color name="catalog_item_detail_indicator_inactive">@color/lightGray</color>
    <color name="catalog_item_detail_indicator_active">@color/off_white</color>

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CircleIndicatorRecyclerViewDecoration.java [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+     */
+    private void drawActiveCircle(Canvas canvas, float initialXPosition, float initialYPosition,
+                                  int activePosition, float swipeProgress) {
+        if (activePosition < 0) { return; }
+        paint.setColor(activeColor);
+
+        float itemWidth = circleDiameter + circlePadding;
+        float highlightStart = initialXPosition + itemWidth * activePosition;
+        float offset = 0F;
+        if (swipeProgress > 0.001F && swipeProgress < 1.001F) {
```
[the remark this answers is not in the document]
Comment (Alvtag) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CircleIndicatorRecyclerViewDecoration.java]: will change to float-vs-integer comparison 
`if (swipeProgress > 0 && swipeProgress < 1) {`

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/adapters/ImageCarouselRecyclerViewAdapter.java (hunk 1 above)
Comment (Alvtag) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/adapters/ImageCarouselRecyclerViewAdapter.java]: changed data to immutable

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+
+    public void setImageURLList(List<URL> imageUrls) {
+        adapter.setData(imageUrls);
+    }
+
+    private void initView() {
+        recyclerView = (RecyclerView) inflate(getContext(), R.layout.view_carousel, null);
+        LayoutParams params = new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.WRAP_CONTENT);
+        recyclerView.setLayoutParams(params);
```
[the remark this answers is not in the document]
Comment (Alvtag) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java]: // During initView we don't have the images loaded yet; this leads renderer to assign a very narrow space
        // for the recyclerView. Re-setting the layout params here fixed the issue.

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java (hunk 2 above)
Comment (Alvtag) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java]: updated - please re-review

File: traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    public CarouselView(Context context) {
+        super(context);
+        initView();
+    }
+
+    public void setAdapter(RecyclerView.Adapter adapter) {
+        recyclerView.setAdapter(adapter);
+    }
+
+    public void setLayoutManager(RecyclerView.LayoutManager layoutManager) {
```
Comment (ata-n) [inline traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java]: is there a need for this?

Comment (Alvtag) [reply traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/widgets/CarouselView.java]: replaced with hardcoded horizontal LLM

**Pull request merged**